vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp

```diff
@@ -74,17 +83,45 @@ $( document ).ready(function() {
 	
 	$("#createSpaceLinkBtn").click(function(e) {
 		var payload = {};
+		var posX = $("#bgImage").position().left;
+        var posY = $("#bgImage").position().top;
 		payload["x"] = storeX;
 		payload["y"] = storeY;
 		payload["rotation"] = $("#spaceLinkRotation").val();
 		payload["linkedSpace"] = $("#linkedSpace").val();
+		payload["spaceLinkLabel"] = $("#spaceLinkLabel").val();
 		payload["type"] = $("#type").val();
 		$("#arrow").remove();
 		$.post("<c:url value="/staff/space/${space.id}/spacelink?${_csrf.parameterName}=${_csrf.token}" />", payload, function(data) {
+
 			// TODO: show success/error message
-		});
+		}); 
+	    
+	    if (payload["type"] == "ALERT") {
+			var icon = $('<div class="alert alert-primary" role="alert"><p>'+payload["spaceLinkLabel"]+'</p>');
+		} else {
+			var icon = $('<span data-feather="navigation-2" class="flex"></span><p class="label-visibility">'+payload["spaceLinkLabel"]+'</p>'); 
+		}	    
+	    icon.css('position', 'absolute');
+	    icon.css('left', storeX + posX);
+	    icon.css('top', storeY + posY);
+	    icon.css('fill', 'red');
+	    icon.css('color', 'red');
+	    icon.css('transform', 'rotate(' +$('#spaceLinkRotation').val()+ 'deg)');
+	    icon.css('font-size', "10px");
+	    
+	    $("#space").append(icon);
+	    feather.replace();
 		$("#bgImage").on("click", function(e){});
-		$("#createSpaceLinkAlert").hide();	
+
+		$("#createSpaceLinkAlert").hide();
+		
+		$(".label-visibility").css({
+	        'transform': 'rotate(0deg)',
+	        'left': payload["x"] + posX - 10,
+	        'top': payload["y"] + posY + 16,
+	        'color': 'white'
```

Comment:
white labels do not show up on shite background, for now let's make them red like the arrows

Reply:
okay! About retesting, do you want the alert div to be reflected on the image as soon as the type is selected as 'alert'?

Reply:
that would definitely make sense. "retest this please" is meant for Jenkins though ;)